File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int res = 0;
    int cnt = 0;
    public int kthSmallest(TreeNode root, int k) {
        helper(root,k);
        return res;
    }

    public void helper(TreeNode root, int k) {
        if (root == null) {
            return ;
        }
        helper(root.left,k);
        cnt++;
        if (cnt == k) {
            res = root.val;
        }
        helper(root.right,k);

        
    }
}

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    cnt = 0
    ans = 0
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.helper(root, k)
        return self.ans
    def helper(self, root, k):
        if not root: 
            return None
        self.helper(root.left,k)
        self.cnt += 1
        if self.cnt == k:
            self.ans = root.val
        else:
            self.kthSmallest(root.right,k)